test/e2e/targets/awssqs/main.go

```diff
@@ -163,6 +163,51 @@ var _ = Describe("AWS SQS target", func() {
 				})
 			})
 		})
+
+		When("the CloudEvent context is discarded", func() {
+			BeforeEach(func() {
+				By("creating an AWSSQSTarget object with discardCEContext enabled", func() {
+					trgt, err := createTarget(trgtClient, ns, "test-",
+						withARN(queueARN),
+						withCredentials(awsSecret.Name),
+						withDiscardCEContext(),
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					trgt = ducktypes.WaitUntilReady(f.DynamicClient, trgt)
+
+					trgtURL = ducktypes.Address(trgt)
+					Expect(trgtURL).ToNot(BeNil())
+				})
+			})
```

Comment:
The way this test suite is structured will lead to two `AWSSQSTarget` objects being created inside the same namespace.

```go
Context("a target is deployed", func() {

    BeforeEach(func() {

        // first object created here

    })

    When("the CloudEvent context is discarded", func() {

        BeforeEach(func() {

            // second object created here

        })
    })
})
```

To avoid that, we should move the first object creation inside its own nested `Context/When/Describe`:

```go
Context("a target is deployed", func() {

    BeforeEach(func() {

        // queue creation can stay here, since all tests require it

    })

    When("the spec contains default settings", func() {

        BeforeEach(func() {

            // only one object created here

        })
    })

    When("the CloudEvent context is discarded", func() {

        BeforeEach(func() {

            // only one object created here

        })
    })
})
```